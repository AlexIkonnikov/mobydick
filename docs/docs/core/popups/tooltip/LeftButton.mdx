import LeftButton from '../assets/LeftButton.png';

<img src={LeftButton} width="50%"/>

Reusable LeftButton component for Tooltip

## Example

```tsx
import {useRef} from 'react';

import {
  Button,
  createStyles,
  IPlacement,
  IPosition,
  TooltipBase,
  Typography,
  usePopups,
  useStyles,
  View,
} from '@lad-tech/mobydick-core';

export const TooltipWidget = () => {
  const ref = useRef(null);
  const [styles] = useStyles(styleFn);
  const {openPopup} = usePopups();

  return (
    <View style={styles.container}>
      <Typography font={'Regular-Primary-H5'}>Tooltip</Typography>
      <View ref={ref} style={styles.ref}>
        <Typography>Ref component</Typography>
      </View>
      <Button
        text={'Tooltip'}
        onPress={() =>
          openPopup({
            Content: props => (
              <TooltipBase
                position={IPosition.bottom}
                placement={IPlacement.center}
                refCurrent={ref}
                {...props}>
                <TooltipBase.Title title={'Tooltip title'} />
                <TooltipBase.LeftButton text={'Left button'} />
              </TooltipBase>
            ),
          })
        }
      />
    </View>
  );
};

const styleFn = createStyles(({spaces, colors}) => ({
  container: {gap: spaces.Space8},
  ref: {
    alignItems: 'center',
    padding: spaces.Space8,
    backgroundColor: colors.BgAccent,
  },
}));
```

## `Props`

### [`IButton Props`](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/cta/components/Button/types.ts#L35)

Inherits [IButton Props](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/cta/components/Button/types.ts#L35)


