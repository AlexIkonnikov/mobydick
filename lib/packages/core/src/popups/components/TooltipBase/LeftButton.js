"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Button_1 = __importDefault(require("../../../cta/components/Button/Button"));
const types_1 = require("../../../cta/components/Button/types");
const useTheme_1 = __importDefault(require("../../../styles/hooks/useTheme"));
const LeftButton = props => {
    const { spaces } = (0, useTheme_1.default)();
    return (<Button_1.default size={types_1.IButtonSize.small} style={{ alignSelf: 'flex-start', marginVertical: spaces.Space8 }} {...props}/>);
};
exports.default = LeftButton;
