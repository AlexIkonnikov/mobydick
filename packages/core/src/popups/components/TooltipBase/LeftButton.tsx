import React, {FC} from 'react';

import Button from '../../../cta/components/Button/Button';
import {IButtonProps, IButtonSize} from '../../../cta/components/Button/types';
import useTheme from '../../../styles/theme/hooks/useTheme';

const LeftButton: FC<IButtonProps> = props => {
  const {spaces} = useTheme();

  return (
    <Button
      size={IButtonSize.small}
      style={{alignSelf: 'flex-start', marginVertical: spaces.Space8}}
      {...props}
    />
  );
};

export default LeftButton;
