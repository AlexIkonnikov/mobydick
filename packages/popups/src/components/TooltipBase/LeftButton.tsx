import {Button, IButtonProps, IButtonSize} from '@npm/mobydick-cta';
import React, {FC} from 'react';
import {useTheme} from '@npm/mobydick-styles';

const LeftButton: FC<IButtonProps> = props => {
  const {spaces} = useTheme();

  return (
    <Button
      size={IButtonSize.small}
      style={{alignSelf: 'flex-start', marginVertical: spaces.Space8}}
      {...props}
    />
  );
};

export default LeftButton;
