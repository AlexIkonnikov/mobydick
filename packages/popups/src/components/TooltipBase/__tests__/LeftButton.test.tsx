import {render} from '@testing-library/react-native';
import React from 'react';
import {IButtonSize} from '@npm/mobydick-cta';

import LeftButton from '../LeftButton';

describe('@npm/mobydick-popups/TooltipBase/LeftButton', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(
      <LeftButton size={IButtonSize.fixed} text={'text'} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
