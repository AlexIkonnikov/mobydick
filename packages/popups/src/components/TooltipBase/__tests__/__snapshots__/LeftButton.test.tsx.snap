// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/TooltipBase/LeftButton should renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#2B78EE",
      "borderRadius": 18,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 12,
      "minHeight": 72,
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 27,
          "lineHeight": 33,
        },
        Array [
          Object {
            "color": "#FFF",
            "paddingHorizontal": 9,
          },
          undefined,
        ],
      ]
    }
  >
    text
  </Text>
</View>
`;
